### DISCUSSION TITLE
Freetype delegate issue

### DISCUSSION THREAD

**Discussion by MarliesMaalderink** [ImageMagick/ImageMagick#2328] [Help] [answered]
It appears as if I am missing the text delegates (freetype/fontconfig) and I don't know how to solve it.

I am using ImageMagick 7.0.9-2 on Ubuntu 20.04 (I have been using it for a while but never had to use text) and I try to add some text to an image. I get this error: delegate library support not built-in '/usr/share/fonts/type1/gsfonts/n019003l.pfb' (Freetype)

These are my delegates: Delegates (built-in): jng jpeg png zlib
So the error makes sense since freetype and fontconfig are missing.

Just now I installed freetype, it seems fontconfig is already on my system. But I am not sure how to make IM know that the delegates are there. I am also not sure how I can be sure they are installed correctly. If I try to install them (again) I get this message:
fontconfig-config is already the newest version (2.13.1-2ubuntu3)
freetype2-demos is already the newest version (2.10.1-2)
So I assume they are there. Though I have doubts about the the freetype2-demos (why is it called demos?)

convert -list font does return a list of fonts, including the one the error complains about.

Should I reinstall IM entirely? Or is there a way to add this delegate without reinstalling?

**Comment by snibgo** [comment, marked as answer]
> Delegates (built-in): jng jpeg png zlib

A _built-in_ delegate is one that is linked in to the IM binary. So you can't add a built-in delegate without re-building IM. This needs at least ./configure, make and make install. Before those, you may also need automake and autoconf.

So, if you recently installed freetype, you need to then rebuild IM.

**Reply by MarliesMaalderink** [reply]
I rebuilt IM and now it works fine. Thank you for answering!
